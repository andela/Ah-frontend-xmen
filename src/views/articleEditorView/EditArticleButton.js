/* eslint-disable react/prefer-stateless-function */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import jwt from 'jsonwebtoken';


export class EditButton extends Component {
  render() {
    const decoded = jwt.decode(localStorage.getItem('token'));
    if (decoded) {
      const loggedUser = decoded.user_data.split(' ')[1];
      const hideButton = loggedUser === this.props.username;
      return (
        <div className="mr-2">
          <Link to={`/editor/${this.props.slug}`}>
            <i
              className="fas fa-pen"
              slug={this.props.slug}
              hidden={!hideButton}
              role="button"
              tabIndex="0"
              onKeyPress={() => {}}
              data-tip="Edit the Article"
            />
          </Link>
        </div>
      );
    }
    return null;
  }
}


const mapStateToProps = state => ({
  EditState: state.articleEditorReducer,
});

export default connect(mapStateToProps)(EditButton);
