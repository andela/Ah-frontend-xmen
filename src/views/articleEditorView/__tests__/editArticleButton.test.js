import React from 'react';
import { shallow } from 'enzyme';
import { EditButton } from '../EditArticleButton';


describe('<EditButton />', () => {
  const props = {
    slug: 'jkdjd',
    hidden: 'kdhdhd',
  };
  beforeEach(() => {
    localStorage.setItem('token',
      'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyX2RhdGEiOiJuYWJycmlra0B5YWhvby5jb20gcmljaGFyZCIsImV4cCI6MTU1MTE4NjA2NH0.Mqt5-sPOqR9GSzddLpHWiivNYDqDlIE15QHoqU8Enrk');
  });
  afterEach(() => {
    localStorage.clear();
  });
  it('should render without crashing', () => {
    localStorage.clear();
    const wrapper = shallow(<EditButton {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render without crashing when logged in', () => {
    const wrapper = shallow(<EditButton {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
