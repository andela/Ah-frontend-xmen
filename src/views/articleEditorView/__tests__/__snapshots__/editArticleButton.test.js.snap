// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditButton /> should render without crashing 1`] = `ShallowWrapper {}`;

exports[`<EditButton /> should render without crashing when logged in 1`] = `ShallowWrapper {}`;
